src/geometry/CTiglIntersectionCalculation.cpp

```diff
@@ -101,6 +104,29 @@ namespace
     private:
         size_t hash;
     };
+
+    class HashableProjection
+    {
+    public:
+        HashableProjection(const gp_Pnt& x, const gp_Dir& d)
+            : hash(0)
+        {
+            gp_Vec xvec = gp_Vec(x.XYZ());
+            gp_Vec dvec = gp_Vec(d);
+            double coeff  = -dvec.Dot(xvec)/dvec.Dot(dvec);
+            gp_Vec result = xvec + coeff*dvec;
+            boost::hash_combine(hash,result.X());
```

Comment:
Shouldn't be the code as follows?:
```
hash = boost::hash<double>()(result.X());
boost::hash_combine(hash,result.Y());
boost::hash_combine(hash,result.Z());
```
Else, you are combining a hash with a hash zero, which probably causes issues.